models/evc.py

```diff
@@ -650,15 +650,29 @@ def remove_path_flows(self, path=None, force=True):
             return
 
         dpid_flows_match = {}
-        for dpid, flows in self._prepare_nni_flows(path).items():
+
+        try:
+            nni_flows = self._prepare_nni_flows(path)
+        except Exception as err:
+            log.error(f"Fail to remove NNI failover flows for {self}: {err}")
+            nni_flows = {}
+
+        for dpid, flows in nni_flows.items():
             dpid_flows_match.setdefault(dpid, [])
             for flow in flows:
                 dpid_flows_match[dpid].append({
                     "cookie": flow["cookie"],
                     "match": flow["match"],
                     "cookie_mask": int(0xffffffffffffffff)
                 })
-        for dpid, flows in self._prepare_uni_flows(path, skip_in=True).items():
+
+        try:
+            uni_flows = self._prepare_uni_flows(path, skip_in=True)
+        except Exception as err:
```

Comment:
Preferred to leave a generic exception handler to catch unexpected errors and avoid the handle link down stops in the middle of its execution.